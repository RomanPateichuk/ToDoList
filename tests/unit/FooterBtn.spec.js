import { mount } from "@vue/test-utils";
import FooterBtnComponent from "../../src/components/atoms/FooterBtn.vue";


describe("Тестирование компонента FooterBtn:", () => {
    const wrapper = mount(FooterBtnComponent, {
     propsData:{
        FooterBtnValue: "ALL"
     },
 
    });

    const input = wrapper.find("input")
    test("соответствие снимку", () => {
      expect(wrapper).toMatchSnapshot();
    });
    
    test("Проверка входных параметров", async () => {
      expect(wrapper.find("span").text()).toBe("ALL")
    });
  

    test("при клике вызываются методы 'callFilter'", async () => {
        await input.trigger("click");
        expect(wrapper.emitted().callFilter).toBeTruthy();
        
      });
  });
  
