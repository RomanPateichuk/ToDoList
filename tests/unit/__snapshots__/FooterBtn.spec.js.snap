// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирование компонента FooterBtn: соответствие снимку 1`] = `<label><input type="radio" name="radio-footer"><span>ALL</span></label>`;
